src/composeappmanager.cc

```diff
@@ -197,7 +138,8 @@ bool ComposeAppManager::fetchTarget(const Uptane::Target& target, Uptane::Fetche
     }
 
   } else {
-    for (const auto& pair : cur_apps_to_fetch_and_update_) {
+    const auto target_apps = Target::targetApps(target, cfg_.apps);
```

Comment:
Maybe LiteClient:getTarget should return this "shortListed" target. Then the packagemanger gets one Target and doesn't have all this decision making to do.

Reply:
I've been considering it many times too. I really want to make the package manager "policyless" so it does exactly what it's told to do. So, in this case, LiteClient would do the shortlisting and determining what needs to be updated just once and pass the shortlisted Target to the package manager. One of the things that makes me hesitant to do it is the fact that `compose_apps` is a configuration parameter of the package manager, not LiteClient, conceptually, LiteClient should not make any assumptions about a package manager instance it uses nor "knows" internal details of it. But, we might forget about this concept and do a shortcut here and just try it. Anyway, technically it's possible, and wanted to give it try and see how it will work.

Also, it would be great to log all Target's apps and mark which of the apps is 'on' and 'off' as we do it now, it kind of makes it difficult to implement the shortlisting in such a way that it happens once and that's why returning the shortlisted Target in `LiteClient:getTarget` might be not the best option. Anyway, this WIP and I am still exploring different ways to minimize amount of the source code and make it cleaner and easily readable.

One of the options that look quite appealing to me is to wrap Uptane::Target with a new entity (e.g. Aklite::Target) that would encapsulate the business logic around shortlisting and iterating over Target apps...

Reply:
> One of the options that look quite appealing to me is to wrap Uptane::Target with a new entity (e.g. Aklite::Target) that would encapsulate the business logic around shortlisting and iterating over Target apps...

I've thought about that same thing numerous times in the past. Its probably a good idea.